malleefowl/pe/map.py

```diff
@@ -0,0 +1,158 @@
+import sys
+import json
+import copy
+from multiprocessing import Manager
+
+from pywps import LiteralOutput, ComplexOutput, BoundingBoxOutput
+from pywps import ComplexInput
+from pywps import Format, get_format
+from owslib.wps import Input, Output
+
+from malleefowl.utils import DataWrapper, auto_list
+from malleefowl.pe.task import TaskPE
+from malleefowl.pe.generic_wps import ParallelGenericWPS
+
+
+class ProgressList:
+    """
+    Simple list to track the progress of each execution threads inside a parallel group that is shared by each tasks
+    """
+    def __init__(self):
+        self.list = auto_list(Manager().list(), default_val=0)
+
+    # The progress list must be shared by every worker using this monitor, so avoid deep copy
+    def __deepcopy__(self, memo):
+        return copy.copy(self)
+
+
+class MapPE(TaskPE):
+    """
+   Represent the map part of a parallel group of a workflow
+   The task split an array of data so that they can be processed concurrently by the downstream PEs
+   """
+
+    # Name of the PE input/output
+    MAP_INPUT = 'map_in'
+    MAP_OUTPUT = 'map_out'
```

Comment:
PEP8 would require these variable names to be lower case. If there is a need for variable name mangling we could use __<var_name> .